spec/stake/arithmetic.md

```diff
@@ -0,0 +1,59 @@
+# Fixed-Point Arithmetic for Rate Computation
+
+To compute base reward rate, base exchange rate, validator-specific exchange
+rates, and total validator voting power, we need to carefully perform
+arithmetic to avoid issues with precision and rounding. We use explicitly
+specified fixed-precision arithmetic for this, with a precision of 8 digits.
+This allows outputs to fit in a u64, with all products fitting in the output
+and plenty of headroom for additions.
+
+All integer values should be interpreted as unsigned 64-bit integers, with the
+exception of the validator's commission rate, which is a `u16` specified in
+terms of basis points.
+
+## Base Reward Rate
+
+The base reward is an input to the protocol, and the exact details of how this
+base rate $r_{e}$ is determined is not yet decided. For now, we can assume it is
+derived from the block header. $r_{e}$ is a fixed-precision `u64` integer with
+8 digits of precision.
+
+
+## Base Exchange Rate
+
+The base exchange rate, $\psi(e)$, can be safely computed as follows:
+
+$$\psi(e) = \prod_{0 \leq i < e} \lfloor \frac{(1e8 + r_i)}{1e8} \rfloor $$
+
+
+## Commission Rate from Funding Streams
+
+To compute the validator's commission rate from its set of funding streams, the following calculation can be performed:
+
+$$c_{v,e} = \sum_{0 \leq i < n} (s_{i,e})$$
+
+Where $s$ is the set of rates defined by the funding streams, $s_{i,e}$ being
+the ith rate at epoch $e$.  $s_i$ should be a fixed-precision `u16` integer
+type, specified in terms of [basis
+points](https://en.wikipedia.org/wiki/Basis_point). The resulting $c_{v,e}$ is
+defined in terms of basis points (i.e., four digits of precision). 
```

Comment:
nit: this phrasing is slightly mixing between the units of measurement and the bits of precision; in our case, we're using integer values, so these are connected, but in principle the two concepts are doing different things (you could have only a few digits of precision, but for some arbitrary base quantity)